### DISCUSSION TITLE
Unable to connect Kex and Wsl2 on windows 11

### DISCUSSION THREAD

**Discussion by catpool91** [microsoft/WSL#7689] [Q&A]
Hi, I'm trying to use Kali Linux from Windows Store on windows 11, the steps I follow are the next:
1.- Install Kali Linux wsl2
2.- Run command: sudo apt update
3.- sudo apt install -y kali-win-kex
4.- kex --sl -s
![Captura de pantalla 2021-11-13 152926](https://user-images.githubusercontent.com/21169208/141659569-2a3880b9-dbd7-4a1d-a2c4-0cb66b8c111f.jpg)
Sometimes the message says: "failed to activate virtual core keyboard"

I´m using TinyWall, but I add rules to Windows Firewall.
![Captura de pantalla 2021-11-13 153630](https://user-images.githubusercontent.com/21169208/141659720-8d7f3815-c83d-4346-803b-6a430b981ea4.jpg)

5.- After, I use kex  kill and kex stop.
6.- Run the command: vncserver with this out:
![Captura de pantalla 2021-11-13 154622](https://user-images.githubusercontent.com/21169208/141660086-eaf22721-9e8a-4d27-95c5-7bd27cea8239.jpg)

7.- kex start
8.- kex --win -s
![Captura de pantalla 2021-11-13 154820](https://user-images.githubusercontent.com/21169208/141659953-1181ceb2-e3c1-4d98-b566-91d85d378f36.jpg)

![Captura de pantalla 2021-11-13 155036](https://user-images.githubusercontent.com/21169208/141659996-ff168cb2-ef21-45ef-b5cb-68f67108de49.jpg)

I hope you can help me :)

**Comment by jnu2019058164** [comment]
try enhanced session mode?use command below :  
`kex --esm -i -s`

**Comment by JSignoretti** [comment]
This issue was from 2021, but since I subscribed and got an email ping, here's my workaround: https://github.com/microsoft/WSL/discussions/6108#discussioncomment-4805781

Basically uninstall "Windows Subsystem for Linux." Reinstall using `wsl.exe --update --inbox`
It's an incompatibility with the updated Microsoft Store version of WSL, but the "inbox" version gets past this error. 

P.S. If you use CUDA stuff for password cracking in WSL, you'll have to compile a newer custom kernel. The inbox version comes with an older kernel without CUDA support.
